book/src/todo_app_1.md

```diff
@@ -0,0 +1,231 @@
+# Building a Simple To-Do App
+
+After we have learned so many concepts, it is finally time to put them into practice.
+We are going to build a To-Do app!
+
+For now, we would already be satisfied with a minimal version.
+An entry to input new tasks and a list view to display them will suffice.
+Something like this:
+
+<div style="text-align:center"><img src="img/todo_app_1.png" /></div>
+
+This mockup can be described by the following composite template.
+
+<span class="filename">Filename: listings/todo_app/1/window/window.ui</span>
+
+```xml
+{{#rustdoc_include ../listings/todo_app/1/window/window.ui}}
+```
+
+In order to use the composite template, we create a custom widget.
+The `parent` is `gtk::ApplicationWindow`, so we inherit from it.
+As usual, we have to list all [ancestors](https://docs.gtk.org/gtk4/class.ApplicationWindow.html#ancestors) and [interfaces](https://docs.gtk.org/gtk4/class.ApplicationWindow.html#implements) apart from `GObject` and `GInitiallyUnowned`.
```

Comment:
Don't we only need to list interfaces that we use?

Reply:
Yes, but I feel it is more complicated to think about which one you use, than just list all of them.
Not sure if I should mention this detail or if it just complicates things.